server/services/v1/database/secondary_indexer.go

```diff
@@ -0,0 +1,489 @@
+		if row.stub && stubs[row.Name()] {
+			continue
+		} else {
+			stubs[row.Name()] = true
+		}
+
+		indexKey := q.buildIndexKey(row, primaryKey)
+		indexKeys = append(indexKeys, indexKey)
+
+		if val, ok := rowCounts[row.Name()]; ok {
+			rowCounts[row.Name()] = val + 1
+		} else {
+			rowCounts[row.Name()] = 1
+		}
+
+		sizeIncrease += int64(len(indexKey.SerializeToBytes()))
+		if val, ok := rowSizes[row.name]; ok {
+			rowSizes[row.Name()] = val + sizeIncrease
+		} else {
+			rowSizes[row.Name()] = sizeIncrease
+		}
+	}
+	return indexKeys, rowSizes, rowCounts
+}
+
+func (q *SecondaryIndexer) indexField(doc []byte, fieldName string, dataType schema.FieldType, pos int, keyPath ...string) (*IndexRow, error) {
+	if dataType == schema.ByteType {
+		return nil, fmt.Errorf("do not index byte field %s", fieldName)
+	}
+	val, dt, _, err := jsonparser.Get(doc, keyPath...)
+	if dt == jsonparser.NotExist {
+		return nil, err
+	}
+
+	row, err := newIndexRow(dataType, q.collation, fieldName, val, pos, false)
+	if err != nil {
+		return nil, err
+	}
+
+	return row, nil
+}
+
+func (q *SecondaryIndexer) indexNestedField(doc []byte, topField string, pos int) []IndexRow {
+	var indexedFields []IndexRow
+	processor := func(key []byte, value []byte, dt jsonparser.ValueType, offset int) error {
+		fieldType := schema.ToFieldType(dt.String(), "", "")
+		fieldName := topField + "." + string(key)
+
+		switch fieldType {
+		case schema.ArrayType:
+			// Create a stub for a nested array
+			row, _ := newIndexRow(fieldType, q.collation, fieldName, nil, pos, true)
+			indexedFields = append(indexedFields, *row)
+		case schema.ObjectType:
+			q.indexNestedField(value, fieldName, pos)
+		default:
+			row, err := newIndexRow(fieldType, q.collation, fieldName, value, pos, false)
+			if err != nil {
+				log.Err(err).Msgf("Failed to index field name: %s", fieldName)
+				return err
+			}
+			indexedFields = append(indexedFields, *row)
+		}
+
+		return nil
+	}
+	err := jsonparser.ObjectEach(doc, processor)
+	if err != nil {
+		log.Err(err).Msgf("Failed to index field name: %s", topField)
+	}
+	return indexedFields
+}
+
+func (q *SecondaryIndexer) indexArray(doc []byte, field *schema.QueryableField, keyPath []string) []IndexRow {
+	pos := 0
+	var rows []IndexRow
+	processor := func(value []byte, dt jsonparser.ValueType, offset int, err error) {
+		toFieldType := schema.ToFieldType(dt.String(), "", "")
+		switch toFieldType {
+		case schema.ObjectType:
+			indexedFields := q.indexNestedField(value, field.FieldName, pos)
+			rows = append(rows, indexedFields...)
+		case schema.ArrayType:
+			indexedField, _ := newIndexRow(toFieldType, q.collation, field.FieldName, nil, pos, true)
+			rows = append(rows, *indexedField)
+		default:
+			indexedField, err := newIndexRow(field.SubType, q.collation, field.FieldName, value, pos, false)
+			if err != nil {
+				log.Err(err).Msgf("Failed to index field name: %s", field.FieldName)
+				return
```

Comment:
But until we have any way to fix these missing indexes we need to fail the transaction. This will also help us catch all the initial bugs. Once we have some way to fix these then we can log it and skip it.